Data.R

```diff
@@ -1,97 +1,131 @@
+#'   html_document:
+#'    code_folding: hide
+#'    toc: true
+#'    toc_float: true
+#' ---
+#'
 #' Loading R libraries
+#+ init, echo=FALSE, message=FALSE, warning=FALSE
+debug <- 0;
+knitr::opts_chunk$set(echo=debug>-1, warning=debug>0, message=debug>0, max.print=42);
 library(ggplot2)
 library(dplyr)
 library(GGally)
 library(rio)
 library(pander)
 library(maps)
+library(printr)
+options(max.print=42)
+panderOptions('table.split.table',Inf); panderOptions('table.split.cells',Inf);
 #' Create data set
 dat0 <- import("covid19cleanedamericas.csv", stringsAsFactors=F);
 #' base R scatterplot matrix
+#+ scatterplot, cache=TRUE
 select(dat0,-'Country/Region') %>% plot()
 #' show first 10 lines of dat0
-head(dat0) %>% pander(split.tables=Inf)
+head(dat0) 
 #' gg plot scatterplot matrix
+#+ ggplot_splom, cache=TRUE
 select(dat0,-'Country/Region') %>% ggpairs()
 #' how to assign value to object and variable
 foo <- 3
 #' How to select columns 2 through 5:
-dat0[,c(2:5)] %>% head %>% pander(split.tables=Inf)
+dat0[,c(2:5)] %>% head
 #' How to select columns by name:
-dat0[,c('Long','Confirmed','Date')] %>% head %>% pander(split.tables=Inf)
+dat0[,c('Long','Confirmed','Date')] %>% head 
 #' This is equivalent to...
 
-head(dat0[,c('Long','Confirmed','Date')])%>% pander(split.tables=Inf)
+head(dat0[,c('Long','Confirmed','Date')])
 #' To see what type each column is
-sapply(dat0, class) %>% pander(split.tables=Inf)
+sapply(dat0, class) %>% pander
 #' Selecting rows instead of columns:
 #'
 #' The old fashioned (base R) way
-dat0[c(1,3,9),] %>% head %>% pander(split.tables=Inf)
+dat0[c(1,3,9),] %>% head()
 #' The subset() command (also base R)
-subset(dat0,Lat<1&Active>0)
+subset(dat0,Lat<1&Active>0) %>% head()
 #' now using OR and equals
-subset(dat0,(Lat<1&Active>0)|`Country/Region`=="Brazil") %>% pander(split.tables=Inf)
+subset(dat0,(Lat<1&Active>0)|`Country/Region`=="Brazil") %>% head()
 #' using OR and excluding
-subset(dat0,(Lat<1&Active>0)|`Country/Region`!="Brazil") %>% pander(split.tables=Inf)
+subset(dat0,(Lat<1&Active>0)|`Country/Region`!="Brazil") %>% head()
 #' Including 
-subset(dat0,(Lat<1&Active>0)|`Country/Region` %in% c("Chile", "Peru")) %>% pander(split.tables=Inf)
+subset(dat0,(Lat<1&Active>0)|`Country/Region` %in% c("Chile", "Peru")) %>% head()
 #' Exclusion of "in" operator
-subset(dat0,(Lat<1&Active>0)|!`Country/Region` %in% c("Chile", "Peru")) %>% pander(split.tables=Inf)
+subset(dat0,(Lat<1&Active>0)|!`Country/Region` %in% c("Chile", "Peru")) %>% head()
 # vignette('dplyr')(tutorial)
-vignette("dplyr")
-# filter allows you to select a subset of rows in a data frame
-dat0 %>% filter(Confirmed == "1", Active == "1")
-# arrange reorders selected rows
-dat0 %>% arrange(Confirmed, Recovered)
-# puts it in descending order
-dat0 %>% arrange(desc(Lat))
-# slice lets you index rows by their integer location
-dat0 %>% slice(5:10)
-# can use head or tail to start from top or bottom
-dat0 %>% slice_head(n=4)
+#vignette("dplyr")
```

Comment:
If you put `help`, `?`, `vignette(...)`, `View(...)`, `browser()`, `debug()`, or any other inherently interactive commands into your script, comment them out so you can run it in batch mode and render reports without these commands interfering.